FEXCore/Source/Interface/Core/JIT/JIT.cpp

```diff
@@ -875,6 +876,13 @@ CPUBackend::CompiledCode Arm64JITCore::CompileCode(uint64_t Entry, uint64_t Size
     for (size_t i = 0; i < DebugData->GuestOpcodes.size(); i++) {
       const auto& GuestOpcode = DebugData->GuestOpcodes[i];
       auto& RIPEntry = JITRIPEntries[i];
+#if defined(ASSERTIONS_ENABLED) && ASSERTIONS_ENABLED
+      uint64_t HostPCOffset = GuestOpcode.HostEntryOffset - CurrentPCOffset;
+      int64_t GuestRIPOffset = GuestOpcode.GuestEntryOffset - CurrentRIPOffset;
+      LOGMAN_THROW_AA_FMT(HostPCOffset <= std::numeric_limits<uint16_t>::max(), "PC offset too large");
```

Comment:
Did you measure a performance benefit from using the `AA` variant here? Otherwise, `A` should be used by default to avoid communicating ambiguous intent.

Reply:
https://github.com/FEX-Emu/FEX/blob/48c03d747abc0357cfdc45ebcbf0995fce39a152/FEXCore/include/FEXCore/Utils/LogManager.h#L49

Perhaps I misunderstood this comment? It seems to state the opposite

Reply:
That comment is [not backed by data](https://github.com/FEX-Emu/FEX/pull/1840#issuecomment-1179511914) and it's certainly not a good default... I hadn't realized (or forgot?) that the PR that introduced this feature mass-converted existing assertions, so this seems like something to address separately.

Reply:
I'm confused though why assume would be harmful? It seems like you would want to assume the condition holds true? At worst the downside is negligible

Reply:
> I'm confused though why assume would be harmful?

In general, effectively placed compiler hints communicate where hotspots are located and what invariants their efficient execution relies on. If you start placing them arbitrarily, identifying the deliberately placed hints becomes impossible.

More pragmatically, pointless `assume` directives will interfere with debugging of non-assertion builds. Since the compiler generates code as-if the condition were true, a debugger will print wrong variable values or have erratic source stepping behavior. Even [debug printfs](https://godbolt.org/z/5oox8jn6h) will print misleading information.

Reply:
Right, I just am uncertain that any such hot paths where this would realistically have an observable impact exist in fex, and as such I think it might be preferable to remove this all together without anything to the contrary?

Reply:
> it might be preferable to remove this all together without anything to the contrary?

That matches my opinion, but I didn't want to bother this PR with that conversation. Sorry if I was unclear that it's not crucial right now, since I only realized it was already in widespread use after my initial comment.